trigger/coap/trigger.go

```diff
@@ -0,0 +1,152 @@
+package coap
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/dustin/go-coap"
+	"github.com/project-flogo/core/data/metadata"
+	"github.com/project-flogo/core/support/log"
+	"github.com/project-flogo/core/trigger"
+)
+
+var triggerMd = trigger.NewMetadata(&Settings{}, &HandlerSettings{}, &Output{})
+
+func init() {
+	_ = trigger.Register(&CoApTrigger{}, &Factory{})
+}
+
+type Factory struct {
+}
+
+// Metadata implements trigger.Factory.Metadata
+func (*Factory) Metadata() *trigger.Metadata {
+	return triggerMd
+}
+
+// New implements trigger.Factory.New
+func (*Factory) New(config *trigger.Config) (trigger.Trigger, error) {
+	s := &Settings{}
+	err := metadata.MapToStruct(config.Settings, s, true)
+	if err != nil {
+		return nil, err
+	}
+
+	return &CoApTrigger{id: config.Id, settings: s}, nil
+}
+
+// Trigger CoAp trigger struct
+type CoApTrigger struct {
+	server   *Server
+	settings *Settings
+	id       string
+	logger   log.Logger
+}
+
+func (t *CoApTrigger) Initialize(ctx trigger.InitContext) error {
+
+	t.logger = ctx.Logger()
+
+	mux := coap.NewServeMux()
+	//TODO .. is it neccessary?
+	//mux.Handle("/.well-known/core", coap.FuncHandler(t.handleDiscovery))
```

Comment:
yes this is required, for discovery...  https://tools.ietf.org/html/rfc6690#section-4